hazelcast-sql/src/test/java/com/hazelcast/sql/impl/expression/string/InitcapFunctionIntegrationTest.java

```diff
@@ -0,0 +1,122 @@
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package com.hazelcast.sql.impl.expression.string;
+
+import com.hazelcast.sql.SqlColumnType;
+import com.hazelcast.sql.SqlErrorCode;
+import com.hazelcast.sql.impl.expression.SqlExpressionIntegrationTestSupport;
+import com.hazelcast.sql.support.expressions.ExpressionValue;
+import com.hazelcast.sql.support.expressions.ExpressionValue.ByteVal;
+import com.hazelcast.sql.support.expressions.ExpressionValue.CharacterVal;
+import com.hazelcast.sql.support.expressions.ExpressionValue.IntegerVal;
+import com.hazelcast.sql.support.expressions.ExpressionValue.LongVal;
+import com.hazelcast.sql.support.expressions.ExpressionValue.ShortVal;
+import com.hazelcast.sql.support.expressions.ExpressionValue.StringVal;
+import com.hazelcast.test.HazelcastParallelClassRunner;
+import com.hazelcast.test.annotation.ParallelJVMTest;
+import com.hazelcast.test.annotation.QuickTest;
+import org.junit.Test;
+import org.junit.experimental.categories.Category;
+import org.junit.runner.RunWith;
+
+import java.math.BigDecimal;
+import java.math.BigInteger;
+
+import static com.hazelcast.sql.support.expressions.ExpressionValue.BigDecimalVal;
+import static com.hazelcast.sql.support.expressions.ExpressionValue.BigIntegerVal;
+import static com.hazelcast.sql.support.expressions.ExpressionValue.DoubleVal;
+import static com.hazelcast.sql.support.expressions.ExpressionValue.FloatVal;
+import static com.hazelcast.sql.support.expressions.ExpressionValue.ObjectVal;
+
+@RunWith(HazelcastParallelClassRunner.class)
+@Category({QuickTest.class, ParallelJVMTest.class})
+public class InitcapFunctionIntegrationTest extends SqlExpressionIntegrationTestSupport {
+    @Test
+    public void test_column() {
+        checkColumn(new StringVal(), null);
+        checkColumn(new StringVal().field1("fiRst"), "First");
+        checkColumn(new StringVal().field1("fiRst seCond"), "First Second");
+
+        checkColumn(new CharacterVal().field1('a'), "A");
+
+        checkColumn(new ByteVal().field1((byte) 100), "100");
+        checkColumn(new ShortVal().field1((short) 100), "100");
+        checkColumn(new IntegerVal().field1(100), "100");
+        checkColumn(new LongVal().field1((long) 100), "100");
+        checkColumn(new BigIntegerVal().field1(new BigInteger("100")), "100");
+        checkColumn(new BigDecimalVal().field1(new BigDecimal("100.5")), "100.5");
+        checkColumn(new FloatVal().field1(100.5f), Float.toString(100.5f));
+        checkColumn(new DoubleVal().field1(100.5d), Double.toString(100.5d));
```

Comment:
We can add these tests:
```java
checkColumn(new DoubleVal().field1(Double.NaN), "Nan");
checkColumn(new DoubleVal().field1(Double.NEGATIVE_INFINITY), "-Infinity");
```
The second one currently fails, but should not.

Reply:
It should not work at all. We currently use too relaxed behavior that should be restricted otherwise. We'd better not dedicate more time to tests that otherwise should not work at all. This behavior will be restricted very soon.